sympy/stats/crv_types.py

```diff
@@ -2195,6 +2196,120 @@ def Normal(name, mean, std):
             MultivariateNormalDistribution, name, mean, std)
     return rv(name, NormalDistribution, (mean, std))
 
+
+#-------------------------------------------------------------------------------
+# Inverse Normal distribution ----------------------------------------------------------
+
+
+class GaussianInverseDistribution(SingleContinuousDistribution):
+    _argnames = ('mean', 'shape')
+
+    @property
+    def set(self):
+        return Interval(0, oo)
+
+    @staticmethod
+    def check(mean, shape):
+        _value_check(shape > 0, "Shape parameter must be positive")
+        _value_check(mean > 0, "Mean must be positive")
+
+    def pdf(self, x):
+        mu, s = self.mean, self.shape
+        return exp(-s*(x - mu)**2 / (2*x*mu**2)) * sqrt(s/((2*pi*x**3)))
+
+    def sample(self):
+        scipy = import_module('scipy')
+        if scipy:
+            from scipy.stats import invgauss
+            return invgauss.rvs(float(self.mean/self.shape), 0, float(self.shape))
```

Comment:
Can you add test(s) for this method? (Travis also runs tests with optional dependencies)
You can refer some tests in lambdify for the format: https://github.com/sympy/sympy/blob/master/sympy/utilities/tests/test_lambdify.py#L197-L204.

Reply:
Thanks. Please review it.

Reply:
We are supposed to deprecate `sample` in the future. It's part of relinking stats to probabilistic language frameworks like PyMC3.